libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -510,3 +737,96 @@ void ContractLevelChecker::checkBaseABICompatibility(ContractDefinition const& _
 		);
 
 }
+
+void ContractLevelChecker::checkAmbiguousOverrides(ContractDefinition const& _contract) const
+{
+	vector<FunctionInfo> baseFuncs;
+
+	for (ContractDefinition const* c: resolveBaseContracts(_contract))
+		for (pair<string, FunctionInfo> const& funcPair: getOverridableFunctions(c))
+		{
+			auto baseFunc = funcPair.second.funcDef;
+
+			auto baseSearchResult =
+				find_if(
+					baseFuncs.cbegin(),
+					baseFuncs.cend(),
+					[&] (auto const& f) {
+						return hasEqualNameAndParameters(*(f.funcDef), *baseFunc);
+					}
+				);
+
+			if (baseSearchResult == baseFuncs.cend())
+			{
+				baseFuncs.emplace_back(funcPair.second);
+				continue;
+			}
+			else if (baseSearchResult->contract == funcPair.second.contract)
+				continue;
+
+			auto const& defFuncs = _contract.definedFunctions();
+			auto const derivedSearchresult = find_if(
+				defFuncs.cbegin(),
+				defFuncs.cend(),
+				[&] (auto const& f) {
+					return hasEqualNameAndParameters(*f, *baseFunc);
```

Comment:
The type of `f` is `auto const&`, but you use `*` - what is really happening here? These are things that are very confusing if automatic type deduction is used.